lib/orderbook/OrderBook.ts

```diff
@@ -13,17 +13,25 @@ import { Models } from '../db/DB';
 import Swaps from '../swaps/Swaps';
 import { SwapDealRole } from '../types/enums';
 import { CurrencyInstance, PairInstance, CurrencyFactory } from '../types/db';
-import { Pair, OrderPortion, OrderIdentifier } from '../types/orders';
+import { Pair, OrderIdentifier, StampedOwnOrder, OrderPortion, StampedPeerOrder } from '../types/orders';
+import { PlaceOrderResult } from '../types/orderBook';
 
 interface OrderBook {
   on(event: 'peerOrder.incoming', listener: (order: orders.StampedPeerOrder) => void): this;
   on(event: 'peerOrder.invalidation', listener: (order: orders.OrderPortion) => void): this;
-  /** Adds a listener to be called when all or part of a local order is filled. */
+  on(event: 'peerOrder.filled', listener: (order: orders.OrderPortion) => void): this;
+  /** Adds a listener to be called when all or part of a local order is swapped. */
+  on(event: 'ownOrder.swapped', listener: (order: orders.OrderPortion) => void): this;
   on(event: 'ownOrder.filled', listener: (order: orders.OrderPortion) => void): this;
```

Comment:
I wanted to continue the approach you started, where we use different names for different events, although all of them could have been just `added` / removed`.
`ownOrder.filled` is indeed for internal matches, and `ownOrder.swapped` is for maker swaps (filled remotely and executed sucessfuly).
I was planning to add documentation for all the events.

Reply:
I kind of like the redundant comments on both `on` and `emit` so that I can just hover over either function and get a quick description of what they do. It's a bit of a hassle to have to update them in both places, but I'm hoping these comments won't change very much going forward.

In thinking about this, I feel it might be better to not have a separate `ownOrder.swapped` method and just reuse `ownOrder.filled`. It seems like we're only using this for the `subscribeRemovedOrders` logic and we are handling the same. From that perspective, it doesn't matter whether an order was filled internally or remotely, right?

Reply:
I will add comments to the listeners.

Yes, we handle `filled` / `swapped` / `invalidated` the same in the regards, so we could have just call them `removed`. But I think there are benefits to the internal separation, mostly for understanding the code base. If we'll keep that, I don't think we should unify `filled` and `swapped` - whatever the names will be, they should differ - one is for local match event, and one if so a successful make swap event.

Reply:
How about having a maker order that gets swapped by a peer emit *both* `ownOrder.swapped` and `ownOrder.filled`? The `ownOrder.swapped` can take the place of `swap.paid` in the `subscribeSwaps` logic, so we won't need to do that maker role check there. That to me seems like maybe the clearest way to structure things.

`swap.paid` - remains as an event that fires any time a swap is settled
`ownOrder.filled` - any time an own order, or a portion of it, is filled whether internally or by remote swap, which ties directly into the `subscribeRemovedOrders` call
`ownOrder.swapped` (with a `SwapResult` parameter) - any time an own order (or portion) is swapped remotely, which ties directly into the `subscribeSwaps` call

Reply:
I’m not sure about this idea - 
* we can already use `ownOrder.swapped` instead of `swap.paid` in `subscribeSwaps`
* emitting two events for one scenario is confusing
* we’ll have 2 use cases (local / remote match) for `ownOrder.filled` without an easy way to differentiate between them in case we'll need to
* in practice we don’t really monitor / announce remote matches. we know about them only as swaps. that’s why I think it’s less correct to add this event

Let me know what you think

Reply:
I’m not sure I understand the difference, because in both `peerOrder.filled` & `ownOrder.filled` the orders are being filled by an own order. In this case, i’m not sure we want to use a more verbose event name. 
Let me know if I missed something.

Reply:
In regards to chaning `subscribeSwaps` - it's requires some changes, because of the different usage of `OrderPortion` / `SwapResult` payload in different streams. We can leave it for a separate PR.

Reply:
I see what you mean, can you just add "by a an own order" to the end of the comments for the `filled` events then? And maybe a TODO for the `subscribeSwaps` change. Thanks.